earthfile2llb/interpreter.go

```diff
@@ -1026,15 +1026,29 @@ func (i *Interpreter) handleEnv(ctx context.Context, cmd spec.Command) error {
 
 var errInvalidSyntax = errors.New("invalid syntax")
 var errRequiredArgHasDefault = errors.New("required ARG cannot have a default value")
+var errGlobalArgNotInBase = errors.New("global ARG can only be set in the base target")
 
 // parseArgArgs parses the ARG command's arguments
 // and returns the argOpts, key, value (or nil if missing), or error
-func parseArgArgs(ctx context.Context, cmd spec.Command) (argOpts, string, *string, error) {
+func parseArgArgs(ctx context.Context, cmd spec.Command, isBaseTarget bool, explicitGlobal bool) (argOpts, string, *string, error) {
 	opts := argOpts{}
 	args, err := flagutil.ParseArgs("ARG", &opts, getArgsCopy(cmd))
 	if err != nil {
 		return argOpts{}, "", nil, err
 	}
+	if opts.Global {
+		// since the global flag is part of the struct, we need to manually return parsing error if it's used while the feature flag is off
+		if !explicitGlobal {
+			return argOpts{}, "", nil, errors.New("unknown flag `global'")
```

Comment:
```suggestion
			return argOpts{}, "", nil, errors.New("unknown flag --global")
```

Reply:
the error that is produced by the parser does not include `--`, so I was just trying to be consistent (including the backtick and the single quote)

Reply:
Oh I see - I think `--global` is better. The parser is sometimes a little weird anyway.